### ISSUE TITLE
Experiencing the same issue of "TypeError:Object(...) is not a function"

### ISSUE THREAD

**Issue by bailanyuan** [orizens/ngx-infinite-scroll#281]
## Expected Behavior
Got the infinite scroll working. I should see the console.log message in the console when doing this:
 onScrollDown() {
    console.log('down');
}

## Actual Behavior
There's no message printing in the console. The onScrollDown doesn't work anymore. The error message in the console is as following:
![image](https://user-images.githubusercontent.com/39800672/40784982-367f56ea-64b5-11e8-8e34-7d9d50671dd2.png)
The html file used is as follwing:
![image](https://user-images.githubusercontent.com/39800672/40785057-76a77ae0-64b5-11e8-9cf7-b39c48d5d389.png)

## Possible Solution
1. This problem might be related to version conflict. I used the dependency before and it works perfectly. The difference is that the ng version has been changed from 0.0.0 to 1.1.5. After updating the ng version, it doesn't work.
I compared the package.json files and those are the things that is different:
![image](https://user-images.githubusercontent.com/39800672/40785760-5fbea7ac-64b7-11e8-82d4-28a2a2f64cfc.png)
![image](https://user-images.githubusercontent.com/39800672/40785788-742c9d70-64b7-11e8-80db-dc370edb9501.png)

2. The version of the dependency I used has also changed from 0.8.4 to 6.0.0.

### Context
Trying to make the infinite scroll work with the data-grid component. It works until the ng version is updated.

### Your Environment

- [ ] Version used:  as above
- [ ] Browser Name and version: latest chrome: Version 66.0.3359.181 (Official Build) (64-bit)
- [ ] (Optional) Operating System and version (desktop or mobile): Windows desktop

**Comment by agscheidle** [comment]
+1

**Comment by agscheidle** [comment]
@bailanyuan I experienced the same before I realized that per my Angular and rxjs versions, I was trying to use to new a version of ngx-infinite-scroll for my angular version. 

I was at Angular ^5.0.0 and rxjs ^5.5.2. It clearly says on the main page "For Angular 4 < v =< 5.5 - use version ngx-infinite-scroll@0.8.4", so with great shame and guilt for having missed that, I installed that and viola.

I resolved it after remembering that rxjs imports can cause weird errors when there are a mix of importing the same augmentations/imports from different locations - ie  `rxjs` vs `rxjs/observable`, etc. Of course differing versions would likely be an issue also - so this package using rxjs 6 might not play so nicely with older versions of rxjs in apps.
